.github/workflows/docs-style.yml

```diff
@@ -0,0 +1,40 @@
+name: Documentation style
+on:
+  push:
+    branches:
+      - master
+      - 0.x
+      - 1.x
+  pull_request:
+    types: [opened, reopened, synchronize, labeled, unlabeled, ready_for_review]
+jobs:
+  releasenotes:
+    name: Generate and check release notes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Install Dependencies
+        run: pip install riot
+
+      - name: Lint release notes
+        run: riot run reno lint
+
+      - name: Generate release notes
+        run: |
+          mkdir -p docs/_build/
+          riot run reno report | tee docs/_build/release_notes.rst
```

Comment:
take a look at `scripts/build-docs`, there is some of this sort of built-in, especially since our spell checker and such for sphinx. I wonder if we should move away from that for linting/checking and only use this stuff?

also.... this won't generate all the release notes, just the ones for the current commit/branch, which could be weird. (just double check this is building the correct release notes)

Reply:
I'm getting the full release notes as this doesn't use the sphinx directive.

And, yes I agree we should consider in place of spell checking with sphinx.